### PR TITLE
node: Use in-mem keyring for tests and clean up test redundancy

### PR DESCRIPTION
**Pull request by renaynay**
This PR:
* creates a `WithCustomKeyring` option to override the default keyring constructed by the node and uses that option to override the keyring with an in-memory keyring for testing purposes inside all unit tests using a `TestNode`; and
* cleans up a lot of redundancy in tests inside the `node` pkg

### CODE REVIEW

**Review by vgonkivs — APPROVED** [review]

**Comment by vgonkivs** [comment]
LGTM 🚀

File: node/settings.go [hunk 1]
```diff
@@ -78,3 +79,11 @@ func WithHeaderConstructFn(construct header.ConstructFn) Option {
 		sets.opts = append(sets.opts, fx.Replace(construct))
 	}
 }
+
+// WithCustomKeyringSigner overrides the default keyring signer constructed
+// by the node.
+func WithCustomKeyringSigner(signer *apptypes.KeyringSigner) Option {
```
Comment (Wondertan) [inline node/settings.go:85]: nit: With...Option already assumes customizability, so `Custom` is redundant, or we should add Custom to each existing options for consistency

File: node/testing.go [hunk 2]
```diff
@@ -21,9 +23,23 @@ func MockStore(t *testing.T, cfg *Config) Store {
...
+		WithNetwork(params.Private),
+		WithCustomKeyringSigner(InMemKeyringSigner(t)),
+	)
 	store := MockStore(t, DefaultConfig(tp))
 	nd, err := New(tp, store, opts...)
 	require.NoError(t, err)
 	return nd
 }
+
+func InMemKeyringSigner(t *testing.T) *apptypes.KeyringSigner {
```
Comment (Wondertan) [inline node/testing.go:38]: Let's call it `TestKeyringSigner`

**Review by Wondertan — APPROVED** [review]

File: node/settings.go (hunk 1 above)
Comment (renaynay) [reply node/settings.go:85]: kk will remove Custom, np

File: node/testing.go (hunk 2 above)
Comment (renaynay) [reply node/testing.go:38]: yup

**Pull request merged**